docs/how-to/upgrade/collect-operator-v1-information.sh

```diff
@@ -0,0 +1,532 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+if [ -n "$V" ]; then
+	set -x
+fi
+
+SCRIPT_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+cd "$SCRIPT_DIR/../../.."
+
+die() {
+	echo "$@" >&2
+	exit 1
+}
+
+format_patch() {
```

Comment:
This script would be easier to understand and review if these functions documented what they expect on stdin, what they generate on stdout, and what the parameters are. For the last part, naming the parameters with `local kind="$1"` would do.

Is it worth it?

Reply:
I added some documentation. Not sure how helpful it is. I'm kind of amazed that it works at all, but it actually is quite nice to use :smile: